horace_core/sqw/@gen_sqw_files_job/private/rundata_write_to_sqw_.m

```diff
@@ -44,6 +44,44 @@
 mpi_obj= MPI_State.instance();
 running_mpi = mpi_obj.is_deployed;
 
+% convert NaN values of runid to sensible values
+% if all values are NaN, assign values by file number
+% if some values are NaN, assign values greater than the maximum non-NaN
+% one
+
+% (a) - find largest non-nan run id
+%     - flag existence of any nans
+maxrunid = NaN;
+hasnans = false;
```

Comment:
```suggestion
hasnans = any(cellfun(@(x) isnan(x.run_id), run_files));
```

Reply:
This also wouldn't need to be recalculated and can just be used in the `if` below

Reply:
useful but the present code is clearer